src/plugins/intel_gpu/src/graph/program.cpp

```diff
@@ -613,7 +608,17 @@ void program::post_optimize_graph(bool is_internal) {
     reorder_factory rf;
     layout_optimizer lo;
     set_layout_optimizer_attributes(lo);
-    apply_opt_pass<post_optimize_weights>(rf);
+
+    bool optimize_data = _config.get_property(ov::intel_gpu::optimize_data);
+
+    if (!is_internal) {
+        apply_opt_pass<reorder_transfer>();
+
+        if (optimize_data) {
+            apply_opt_pass<fuse_constant_transposes>();
```

Comment:
Does reorder_transfer and fuse_constant_transposes need to happen on post-optimize stage? Can't reorder be transfered before fusions, and fuse_constant_transposes be a part of prepare_primitive_fusing pass?